packages/dds/tree/src/shared-tree-core/transactionEnricher.ts

```diff
@@ -13,6 +14,11 @@ export class TransactionEnricher<TChange> {
 	private readonly rebaser: ChangeRebaser<TChange>;
 	private readonly enricher: ChangeEnricherReadonlyCheckout<TChange>;
 	private readonly transactionCommits: GraphCommit<TChange>[] = [];
+	/**
+	 * The number of commits before the start of each active transaction scope.
+	 * Index 0 is used for the outermost transaction scope, the number of commits before it will always be 0.
```

Comment:
Er, so is this saying that the first element in the array, if present, is always 0? Or, is it saying that the number at index 0 is present, but meaningless? Or something else?

Slightly more formal language could be helpful, e.g.:

"For a stack of _n_ nested transactions _t0 ... tn_, this array has a length of _n_ where the value at index _i_ is the number of commits within transaction _ti-1_."

Or something like that.

Reply:
> is this saying that the first element in the array, if present, is always 0?
Yes. How about:
For a stack of `n` transaction scopes, the array will contain `n` integers, where the integer at index `i` is the number of commits made before the start of the `i`th transaction scope.

Reply:
Sounds good to me. I'd add a clause "(therefore, the first element in the array, if present, is always 0)" - it's not strictly necessary, but it will let the reader double check their understanding.